packages/react/checkbox/src/Checkbox.tsx

```diff
@@ -57,10 +57,11 @@ const Checkbox = React.forwardRef((props, forwardedRef) => {
   } = props;
   const [button, setButton] = React.useState<HTMLButtonElement | null>(null);
   const composedRefs = useComposedRefs(forwardedRef, (node) => setButton(node));
-  const buttonSize = useSize(button);
   const labelId = useLabelContext(button);
   const labelledBy = ariaLabelledby || labelId;
   const hasConsumerStoppedPropagationRef = React.useRef(false);
+  // We set this to true by default so that events bubble to forms without JS (SSR)
+  const isFormControl = button ? button.closest('form') : true;
```

Comment:
We could even do:
```suggestion
  const isFormControl = button ? Boolean(button.form) : true;
```
because controls will have a `form` property associated to them when inside a form.

Reply:
I did exactly this initially, but `Slider` isn't a form control, it's a `span` so I went with this to keep the two approaches consistent. Should I do it differently here?